### PR TITLE
Open api generation

### PR DESCRIPTION
**Pull request by juanpedromoreno**
## What this does?

This PR resumes the work by @BeniVF , providing support for code generation from [Open API specifications](https://swagger.io/specification/).

## Checklist

- [x] Reviewed the diff to look for typos, println and format errors.
- [x] Updated the docs accordingly.

### CODE REVIEW

**Comment by juanpedromoreno** [comment]
There still some binary issues with `circe`, after the recent build upgrade (previously was working). 

```bash
[info] [warn] There may be incompatibilities among your library dependencies; run 'evicted' to see detailed eviction warnings.
[info] [error] java.lang.NoSuchMethodError: sbt.package$.fileJsonFormatter()Lsjsonnew/JsonFormat;
[info] [error] 	at higherkindness.mu.rpc.idlgen.IdlGenPlugin$autoImport$.srcGenSourceDirs$lzycompute(IdlGenPlugin.scala:60)
[info] [error] 	at higherkindness.mu.rpc.idlgen.IdlGenPlugin$autoImport$.srcGenSourceDirs(IdlGenPlugin.scala:59)
[info] [error] 	at $1a2dcb12ba29963163ce$.root$lzycompute(build.sbt:19)
[info] [error] 	at $1a2dcb12ba29963163ce$.root(build.sbt:11)
[info] [error] 	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
[info] [error] 	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
[info] [error] 	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
```

**Comment by juanpedromoreno** [comment]
Thanks @fedefernandez for the fix! Actually, I think we don't need to add the `build.properties` files as part of the scripted tests... as you pointed out, looks like a bug in sbt.

**Comment by juanpedromoreno** [comment]
... and @BeniVF, thank you for your hard work in this new feature!

File: modules/idlgen/core/src/main/scala/higherkindness/mu/rpc/idlgen/openapi/OpenApiSrcGenerator.scala [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+          val (_, paths) =
+            file.getParentFile
+              .toPath()
+              .asScala
+              .splitAt(resourcesBasePath.iterator().asScala.size + 1) //we need to add one because it is changing the resource path, adding open api
+          val path: Path = Paths.get(paths.map(_.toString()).mkString("/"))
+          val pkg        = packageName(path)
+          pathFrom(path, file).toString ->
+            Seq(
+              // s"//$resourcesBasePath",
```
Comment (fedefernandez) [inline modules/idlgen/core/src/main/scala/higherkindness/mu/rpc/idlgen/openapi/OpenApiSrcGenerator.scala]: What's the purpose of these comments?

Comment (juanpedromoreno) [reply modules/idlgen/core/src/main/scala/higherkindness/mu/rpc/idlgen/openapi/OpenApiSrcGenerator.scala]: I don't think they are needed :) -> 65f95dc

File: modules/idlgen/core/src/main/scala/higherkindness/mu/rpc/idlgen/openapi/OpenApiSrcGenerator.scala [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+    private def getCode[F[_]: Sync](file: File): Nested[F, Option, (String, Seq[String])] =
+      parseFile[F]
+        .apply(file)
+        .map(OpenApi.extractNestedTypes[JsonSchemaF.Fixed])
+        .map { openApi =>
+          pathFrom(Paths.get("."), "foo") ->
+            Seq(
+              model[JsonSchemaF.Fixed].print(openApi),
+              interfaceDefinition.print(openApi),
+              impl.print(PackageName("foo") -> openApi)
```
Comment (BeniVF) [inline modules/idlgen/core/src/main/scala/higherkindness/mu/rpc/idlgen/openapi/OpenApiSrcGenerator.scala]: Ideally, this should require to be configure by sbt options

File: modules/idlgen/core/src/main/scala/higherkindness/mu/rpc/idlgen/openapi/OpenApiSrcGenerator.scala [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+        serializationType: String,
+        options: String*): Option[(String, Seq[String])] =
+      getCode[IO](inputFile).value.unsafeRunSync()
+
+    private def getCode[F[_]: Sync](file: File): Nested[F, Option, (String, Seq[String])] =
+      parseFile[F]
+        .apply(file)
+        .map(OpenApi.extractNestedTypes[JsonSchemaF.Fixed])
+        .map { openApi =>
+          pathFrom(Paths.get("."), "foo") ->
```
Comment (BeniVF) [inline modules/idlgen/core/src/main/scala/higherkindness/mu/rpc/idlgen/openapi/OpenApiSrcGenerator.scala]: deally, this should require to be configure by sbt option

**Comment by juanpedromoreno** [comment]
@BeniVF thanks for the review. I think this is now ready for a second look. Thanks.

**Pull request merged**